redis_session.c

```diff
@@ -581,6 +581,70 @@ redis_session_key(redis_pool_member *rpm, const char *key, int key_len, int *ses
     return session;
 }
 
+/* {{{ PS_CREATE_SID_FUNC
+ */
+PS_CREATE_SID_FUNC(redis)
+{
+    int retries = 3;
+    redis_pool *pool = PS_GET_MOD_DATA();
+
+    if (!pool) {
+#if (PHP_MAJOR_VERSION < 7)
+        return php_session_create_id(NULL, newlen TSRMLS_CC);
+#else
+        return php_session_create_id(NULL TSRMLS_CC);
+#endif
+    }
+
+    while (retries-- > 0) {
+#if (PHP_MAJOR_VERSION < 7)
+        char* sid = php_session_create_id((void **) &pool, newlen TSRMLS_CC);
+        redis_pool_member *rpm = redis_pool_get_sock(pool, sid TSRMLS_CC);
+#else
+        zend_string sid = php_session_create_id((void **) &pool TSRMLS_CC);
+        redis_pool_member *rpm = redis_pool_get_sock(pool, ZSTR_VAL(sid) TSRMLS_CC);
+#endif
+        RedisSock *redis_sock = rpm?rpm->redis_sock:NULL;
+
+        if (!rpm || !redis_sock) {
+            php_error_docref(NULL TSRMLS_CC, E_NOTICE,
+                "Redis not available while creating session_id");
+
+#if (PHP_MAJOR_VERSION < 7)
+            efree(sid);
+            return php_session_create_id(NULL, newlen TSRMLS_CC);
+#else
+            zend_string_release(sid);
+            return php_session_create_id(NULL TSRMLS_CC);
+#endif
+        }
+
+#if (PHP_MAJOR_VERSION < 7)
+        pool->lock_status->session_key = sid;
+#else
+        pool->lock_status->session_key = estrdup($TR_VAL(sid));
```

Comment:
I'm not sure about the `estrdup()`, but if we're ever going to `efree(lock_status->session_key)` I imagine it's OK.